src/scheduler/workers/unpublish/day/changeState/index.ts

```diff
@@ -0,0 +1,36 @@
+import { schedulerLogger } from "scheduler/logger";
+import { ListWithCountryName } from "../../types";
+import { ListJsonData } from "server/models/types";
+import { unpublishListItems } from "./unpublishListItems";
+import { resetAnnualReviewForList } from "./resetAnnualReviewForList";
+import { getMetaForList } from "scheduler/workers/unpublish/getMetaForList";
+import {
+  findListItemsToResetAnnualReview
+} from "scheduler/workers/unpublish/day/changeState/findListItemsToResetAnnualReview";
+
+export async function changeState(list: ListWithCountryName) {
+  const logger = schedulerLogger.child({ listId: list.id, method: "changeState" });
+
+  const jsonData = list.jsonData as ListJsonData;
+  if (!jsonData?.currentAnnualReview) {
+    return;
+  }
+  const meta = getMetaForList(list);
+  if (!meta) {
+    return;
+  }
+
+  const listItems = await findListItemsToResetAnnualReview(list);
+  logger.info(`retrieved list items: ${listItems.length}`);
+  if (listItems.length) {
+    const unpublishedListItemsTasks = await unpublishListItems(listItems.map((listItem) => listItem.id), jsonData.currentAnnualReview.reference);
+    await Promise.allSettled(unpublishedListItemsTasks);
+    logger.info(
+      `${
+        unpublishedListItemsTasks.length ? "Reset" : "Could not reset"
+      } annual review state for list items [${listItems.map((listItem) => listItem.id)}]`
+    );
+  }
```

Comment:
In a scenario where it 'Could not reset annual review state...' should that error be captured elsewhere?

Reply:
Also, I'm guessing if `unpublishedListItemsTasks ` is an empty array the following line:
```js
await Promise.allSettled(unpublishedListItemsTasks);
```
Will still run. Is that okay?

Reply:
It probably should be warned or errored, but it should be retried the next day presumably

Reply:
Thanks for the feedback.  Added extra logging to capture unfulfilled promises when updating state and sending emails to providers and post contacts.  To your point @jenbutongit , if it did error, because this is the day before email, it wouldn't retry the next day as the unpublish event would kick in.  This logic is applicable to the specific day before unpublished only.

Reply:
with you.  Yep, agreed.  Will add extra logging.